[pkg/ottl] Add RemoveXML Converter (#35301)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

- `ParseXML("<HostInfo hostname=\"example.com\" zone=\"east-1\" cloudprovider=\"aws\" />")`

### RemoveXML

> `RemoveXML(target, xpath)`

Reviewer: Since this function returns an error if the string is not parsable xml, can we include something to help users write a where clause that checks this requirement?

For `ParseJSON` users typically do `where IsMatch(body, "\\{")`. What should they do for xml?

Author: `where IsMatch(body, "\\<")` should work about as reliably for xml as the other does for json. I'm not sure how robust either is really since there can of course be other strings which start with these characters. I guess I would slightly prefer to leave these undocumented for now and we can always add `IsJSON` and/or `IsXML` functions later if necessary.

Reviewer: As long as there is some sort of solution that xml users will understand I'm good with that 

My gut says that IsJSON or IsXML will be inefficient, so. I wonder if the decision to error in ParseJSON/XML is correct.  We should stick with it for now and can address later.

The `RemoveXML` Converter returns an edited version of an XML string with selected elements removed.

`target` is a Getter that returns a string. This string should be in XML format.

> If `target` is not a string, nil, or cannot be parsed as XML, `ParseXML` will return an error.

Reviewer: ```suggestion
If `target` is not a string, nil, or is not valid xml, `RemoveXML` will return an error.
```
